testsuite/generate-format/tests/detect_format_change.rs

```diff
@@ -20,8 +20,16 @@ fn analyze_serde_formats() {
             assert_registry_has_not_changed(&corpus.to_string(), path, registry.clone(), expected);
         }
 
-        // Test that the definitions in all corpus are unique.
+        // Test that the definitions in all corpus are unique and pass the linter.
         for (key, value) in registry {
+            assert_eq!(
+                generate_format::lint_lcs_format(&value),
+                Ok(()),
+                "In corpus {}: lint error while analyzing {}",
+                corpus.to_string(),
+                key
+            );
```

Comment:
Will this tell us what the lint error was when it failed?  It looks like it'll tell you where the problem is, but not what the problem is.

Reply:
Yes it will, as part of the explanation why `Err(err) != Ok(())`.